hudi-common/src/main/java/org/apache/hudi/common/model/OverwriteWithLatestAvroPayload.java

```diff
@@ -49,7 +49,7 @@ public OverwriteWithLatestAvroPayload(Option<GenericRecord> record) {
   @Override
   public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload another) {
     // pick the payload with greatest ordering value
-    if (another.orderingVal.compareTo(orderingVal) > 0) {
+    if (another.orderingVal.compareTo(orderingVal) >= 0) {
```

Comment:
Looks good to me. 
@vinothchandar : Can you think of any particular reason why it was done this way?

Reply:
Thanks @nsivabalan , i also found that now `AbstractHoodieLogRecordScanner` always hard code the `orderingVal` to `0`, and then `HoodieMergedLogRecordScanner` invoke the `newRrecord.preCombine(oldRecord)` to merge the records to implement the natural order.

But in my opinion, `newRrecord.preCombine(oldRecord)` has break the semantics of this interface, so i have also fixed the `AbstractHoodieLogRecordScanner` `orderingVal` fetch logic.

Reply:
Finally i found that the `another` should always be the old value.